components/Comment/lib/findComment.ts

```diff
@@ -0,0 +1,54 @@
+import { ID } from "~/config/types/root_types";
+import { Comment } from "./types";
+
+type FoundComment = {
+  comment: Comment;
+  path: Array<ID>;
+};
+
+type findArgs = {
+  id: ID;
+  comment: Comment;
+  path?: Array<ID>;
+};
+
+type findCommentArgs = {
+  id: ID;
+  comments: Comment[];
+};
+
+const _find = ({
+  id,
+  comment,
+  path = [],
+}: findArgs): FoundComment | undefined => {
+  if (comment.id === id) {
+    return {
+      comment,
+      path: [...path, id],
+    };
+  }
+  for (let i = 0; i < comment.children.length; i++) {
+    const child = comment.children[i];
+    const found = _find({ id, comment: child, path: [...path, comment.id] });
+
+    if (found) {
+      return found;
+    }
+  }
+};
+
+const findComment = ({
+  id,
+  comments,
+}: findCommentArgs): FoundComment | undefined => {
+  for (let i = 0; i < comments.length; i++) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find can be a nice functional alternative

Reply:
Thanks for the comment. The reason for the custom find here is the recursive nature of the search operation. Each comment object has `children` which in turn can have their own child comments and so on...

`findComment` is the exported "public" method and the `_find` is supposed to be the helper recursive call